src/backend/env/post.rs

```diff
@@ -1,11 +1,17 @@
-use std::cmp::{Ordering, PartialOrd};
-
 use super::reports::ReportState;
 use super::*;
 use super::{storage::Storage, user::UserId};
-use crate::mutate;
 use crate::reports::Report;
+use crate::{mutate, performance_counter};
+use ic_stable_structures::{StableBTreeMap, Storable};
 use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
+use std::cell::RefCell;
+use std::cmp::{Ordering, PartialOrd};
+
+thread_local! {
+    pub static POSTS: RefCell<Option<StableBTreeMap<PostId, Post, crate::Memory>>> = Default::default();
```

Comment:
Why don't we use a [StableLog](https://docs.rs/ic-stable-structures/latest/ic_stable_structures/log/struct.Log.html) instead of a map, we allocate the same amount of memory for each post, and to retrieve a post we just look at `postID * post size`?

I'm not really convinced about my suggestion I'm just challenging a bit the choices made.

Reply:
Because we have a cache above the stable memory. So posts need to be removed from the archive and put back eventually. I'll use the Log for ledger transaction, but posts are a volatile data. Or am I missing something about the Log structure?

Reply:
One thing that is still not clear to me is what we store in the buckets. I know that in the Taggr wasm we embed a bucket wasm and we store some stuff in them. Do we only store posts there? Or just pictures?

I also realized the Taggr transactions are not in stable logs, why don't we also add them to stable logs?

Reply:
Buckets hold blobs only (images).

>I also realized the Taggr transactions are not in stable logs, why don't we also add them to stable logs? On upgrade, we would just need to re-compute the balances by reading the transactions.

The recomputation already happens right now, it's just all TXs are living in the heap. Changing this will be one of the next PRs.